File: downloadablebookandexercisecode/1256/code_files/Chapter_02_Solving_and_Coding/USACO_2015_dec_b1_fence_painting_brute_force.py
# http://usaco.org/index.php?page=viewproblem2&cpid=567

fin = open('paint.in', 'r')
fout = open('paint.out', 'w')

a, b = map(int, fin.readline().strip().split())
c, d = map(int, fin.readline().strip().split())

ans = 0

# just check every unit of the fence

for i in range(100):
    p = i + 0.5
    if (a < p < b) or (c < p < d):
        ans = ans + 1

fout.write(f'{ans}')

File: downloadablebookandexercisecode/1256/code_files/Chapter_02_Solving_and_Coding/USACO_2015_dec_b1_fence_painting_brute_force.cpp
// http://usaco.org/index.php?page=viewproblem2&cpid=567

#include <iostream>

using namespace std;

int main() {

    freopen("paint.in", "r", stdin);
    freopen("paint.out", "w", stdout);

    int a, b, c, d;
    cin >> a >> b;
    cin >> c >> d;

    int l = 0;

    // just check every unit of the fence

    for (double i = 0.5; i < 100; ++i ) {
        if ( ( i > a && i < b ) || ( i > c && i < d ) ) {
            l++;
        }
    }

    cout << l;

}
